src/DataSet/ObjectDataSet.php

```diff
@@ -158,6 +159,10 @@ final class ObjectDataSet implements RulesProviderInterface, DataWrapperInterfac
      * @var bool Whether an {@see $object} provided rules by implementing {@see RulesProviderInterface}.
      */
     private bool $rulesProvided;
+    /**
+     * @var bool Whether an {@see $object} provided labels by implementing {@see LabelsProviderInterface}.
+     */
+    private bool $labelsProvided;
```

Comment:
No need to cache `instanceof` in a boolean variable, it is fast enough to check directly in `getValidationPropertyLabels()`.